File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test1/TestApplicationComponent.java
package it.droidcon.testingdaggerrxjava.test1;

import dagger.Component;
import dagger.android.AndroidInjectionModule;
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent;
import it.droidcon.testingdaggerrxjava.dagger.UserInteractorModule;
import it.droidcon.testingdaggerrxjava.userlist.UserListModuleDef;
import javax.inject.Singleton;

@Singleton
@Component(modules = {
    UserInteractorModule.class,
    TestStackOverflowServiceModule.class,
    AndroidInjectionModule.class,
    UserListModuleDef.class
})
public interface TestApplicationComponent extends ApplicationComponent {
  void inject(EndToEndTest endToEndTest);
}

File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test1/TestApplicationComponent.kt
package it.droidcon.testingdaggerrxjava.test1

import dagger.Component
import it.droidcon.testingdaggerrxjava.EspressoUserInteractorModule
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent
import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(EspressoUserInteractorModule::class, TestStackOverflowServiceModule::class))
interface TestApplicationComponent : ApplicationComponent {
    fun inject(userListActivityTest: UserListActivityTest)
}
